### ISSUE TITLE
-augment is not augmenting

### ISSUE THREAD

**Issue by dleangen** [bndtools/bnd#3841]
I am trying to augment org.osgi.util.pushstream to also require pushstream (because otherwise, it fails at runtime).

As a test, when LogStreamProvider resolves, this ought to fail, but does not:

```
-augment.eq_log: \
	org.eclipse.equinox.log.stream; \
		capability:='\
			osgi.service;\
				objectClass:List<String>=org.osgi.service.log.stream.LogStreamProvider;\
				effective:=active',\
		requirement:='\
			osgi.wiring.bundle;\
				osgi.wiring.bundle=org.osgi.util.pushstream',\
    	requirement:='\
			osgi.wiring.bundle;\
				osgi.wiring.bundle=poo'
```

**Comment by bjhargrave** [comment]
What you seem to be doing here is confusing. `org.eclipse.equinox.log.stream` should already import the `org.osgi.util.pushstream` package, so you absolutely do not want to also add a Require-Bundle(`osgi.wiring.bundle`) to the `org.osgi.util.pushstream`  bundle. The import package is correct. 

It may be that the `org.eclipse.equinox.log.stream` bundle does not provide a service capability, so augmenting it for the Bnd resolve operation is fine. But also note that augmenting does nothing at runtime. Augmenting only applies to the Bnd resolve operation.

**Comment by dleangen** [comment]
I needed to move on quickly, so I found a different solution (i.e. hardcoding in a different project that is not already released) and no longer have this problem.

But at resolution time (when trying to build my list of runbundles), including `org.eclipse.equinox.log.stream` did not pull in `org.osgi.util.pushstream`, so I was looking for a solution to that.

I reported this because it looked like a bug because the augment for requirement was not working at resolution time (though capability was). If you say this is correct behaviour, then my bad, I must have again misunderstood the documentation, which seems to state the requirement should work.
